src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -65221,6 +65221,53 @@ void M(string? s)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 13));
         }
 
+        [Fact, WorkItem(39736, "https://github.com/dotnet/roslyn/issues/39736")]
+        public void Foreach_TuplesThroughFunction()
+        {
+            var source = @"
+using System.Collections.Generic;
+class Alpha
+{
+    public string Value { get; set; } = """";
+    public override string ToString() => Value;
+}
+
+class C
+{
+    void M()
+    {
+        var items3 = new List<(int? Index, Alpha? Alpha)>()
+        {
+            (0, new Alpha { Value = ""A"" }),
+            (1, new Alpha { Value = ""B"" }),
+            (2, new Alpha { Value = ""C"" })
+        };
+        foreach (var indexAndAlpha in Identity(items3))
+        {
+            var(index, alpha) = indexAndAlpha;
+            index.ToString();
+            alpha.ToString(); // 1
```

Comment:
>alpha.ToString(); [](start = 12, length = 17)

Consider testing assignment to those variables, or displaying their inferred types with `VerifyTypes`